network/rateLimitedTransport.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) 2019-2020 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package network
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"time"
+)
+
+// RateLimitedTransport is a wrapper around the http.Transport that overrides the Dial and DialContext functions.
+// It limits the rate of the outgoing connections to comply with connectionsRateLimitingCount
+type RateLimitedTransport struct {
```

Comment:
you've tried to create some new "hybrid" creature for transport and dialer; Not so sure it's a good idea, since it will make it much harder down the road. Plus, you didn't really initialized the transport so I'm not even sue what this even mean.